static/src/javascripts/projects/facia/modules/ui/live-blog-updates.js

```diff
@@ -63,46 +64,98 @@ const renderBlock = (
         .then(vPosition => {
-            if (vPosition > 0 && vPosition < viewportHeightPx) {
-                setTimeout(
-                    () => {
-                        const $el = bonzo(el);
-                        fastdomPromise.write(() => {
-                            $el.removeClass(
-                                'fc-item__liveblog-blocks__inner--offset'
-                            );
-                        });
-                    },
-                    immediate ? 0 : animateDelayMs
+            const isVisible = vPosition > 0 && vPosition < viewportHeightPx;
+
+            if (isVisible) {
+                timeoutPromise(immediate ? 0 : animateDelayMs).then(() =>
+                    fastdomPromise.write(() =>
+                        el.classList.remove(
+                            'fc-item__liveblog-blocks__inner--offset'
+                        )
+                    )
                 );
+
                 return true;
             }
             return false;
         });
 
-const animateBlocks = (el: Element): void => {
-    maybeAnimateBlocks(el).then(didAnimate => {
-        let animateOnScroll;
-
+const animateBlocks = (el: Element, container: Element): void => {
+    maybeAnimateBlocks(el, container).then(didAnimate => {
         if (!didAnimate) {
-            animateOnScroll = debounce(() => {
-                maybeAnimateBlocks(el, true).then(animated => {
-                    if (animated) {
-                        mediator.off('window:throttledScroll', animateOnScroll);
+            const animateOnScroll = debounce(() => {
+                maybeAnimateBlocks(el, container, true).then(
+                    didAnimateOnScroll => {
+                        if (didAnimateOnScroll) {
+                            mediator.off(
+                                'window:throttledScroll',
+                                animateOnScroll
+                            );
+                        }
                     }
-                });
+                );
             }, animateAfterScrollDelayMs);
 
             mediator.on('window:throttledScroll', animateOnScroll);
         }
     });
+
+    fastdomPromise.write(() => {
+        container.classList.remove('fc-item__liveblog-blocks--hidden');
+        container.classList.add('fc-item__liveblog-blocks--visible');
+    });
+};
+
+const applyUpdate = (
+    container: Element,
+    content: Array<Element>
+): Promise<void> =>
+    fastdomPromise.write(() => {
+        bonzo(container)
+            .empty()
+            .append(content);
+    });
+
+const startUpdate = (
+    container: Element,
+    content: Array<Element>,
+    shouldTransitionOut: boolean
+): Promise<void> => {
+    if (shouldTransitionOut) {
+        container.classList.remove('fc-item__liveblog-blocks--visible');
+        container.classList.add('fc-item__liveblog-blocks--hidden');
+
+        // Use a timeout here so we don't update the DOM until the transition
```

Comment:
Maybe https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/transitionend_event might help?

Reply:
Oh nice - TIL. I'll look into switching to that.

Reply:
Only thing might be browser support - Edge may be an issue, so might not be worth investing too much time!

Reply:
According to https://caniuse.com/#feat=css-transitions support is there for Edge and IE: "Support listed is for transition properties as well as the transitionend event." I've left this as-is for now, but would be worth following up on in a separate PR I think.

Reply:
Yup, makes sense!